admin/acceptance_tests/metadatas_test.py

```diff
@@ -0,0 +1,209 @@
+            ]
+            for m in metadatas]
+        )
+
+    def _post_metadata(self, test_app, url, base_mapping, name, value, status):
+        return test_app.post(
+            url,
+            base_mapping + (
+                ('__start__', 'metadatas:sequence'),
+                ('__start__', 'metadata:mapping'),
+                ('name', name),
+                (self.__metadata_ui_type(test_app, name), value),
+                ('__end__', 'metadata:mapping'),
+                ('__end__', 'metadatas:sequence'),
+            ),
+            status=status)
+
+    def _post_invalid_metadata(self, test_app, url, base_mapping, name, value, error_msg):
+        resp = self._post_metadata(
+            test_app, url, base_mapping, name, value, 200
+        )
+        assert error_msg == \
+            resp.html.select_one(
+                '.item-{} .help-block'.format(self.__metadata_ui_type(test_app, name))
+            ).getText().strip()
+        return resp
+
+    def _base_metadata_params(self, metadatas_test_data):
+        return (
+            ('name', 'new_name'),
+            ('ogc_server_id', metadatas_test_data['ogc_server'].id),
+            ('layer', 'new_wmslayername')
+        )
+
+    def test_invalid_float_metadata(self, test_app, metadatas_test_data):
+        self._post_invalid_metadata(
+            test_app,
+            '/layers_wms/new',
+            self._base_metadata_params(metadatas_test_data),
+            '_float',
+            'number',
+            '"number" is not a number')
+
+    def test_valid_float_metadata(self, test_app, metadatas_test_data):
+        self._post_metadata(
+            test_app,
+            '/layers_wms/new',
+            self._base_metadata_params(metadatas_test_data),
+            '_float',
+            '2.5',
+            302)
+
+    def test_invalid_int_metadata(self, test_app, metadatas_test_data):
+        self._post_invalid_metadata(
+            test_app,
+            '/layers_wms/new',
+            self._base_metadata_params(metadatas_test_data),
+            '_int',
+            'number',
+            '"number" is not a number')
+
+    def test_valid_int_metadata(self, test_app, metadatas_test_data):
+        self._post_metadata(
+            test_app,
+            '/layers_wms/new',
+            self._base_metadata_params(metadatas_test_data),
+            '_int',
+            '2',
+            302)
+
+    def _test_edit_treeitem(self, prefix, item, test_app):
+        resp = self.get(test_app, '{}/{}'.format(prefix, item.id))
+        self._check_metadatas(test_app,
+                              resp.html.select_one('.item-metadatas'),
+                              item.metadatas)
+        resp.form.submit('submit', status=302)
+
+    def test_layer_wms_metadatas(self, metadatas_test_data, test_app):
+        self._test_edit_treeitem('layers_wms', metadatas_test_data['layer_wms'], test_app)
+
+    def test_layer_wmts_metadatas(self, metadatas_test_data, test_app):
```

Comment:
following item.metadatas are empty ?

Reply:
Yes, for the moment, this only checks the sequence is rendered.
We should filter available names depending on treeitem types later.